lib/lndclient/LndClient.ts

```diff
@@ -44,15 +46,14 @@ class LndClient extends BaseClient {
       this.logger.error('could not find lnd macaroon, is lnd installed?');
       this.setStatus(ClientStatus.DISABLED);
     } else {
+      this.uri = `${config.host}:${config.port}`;
       const lndCert = fs.readFileSync(certpath);
-      const credentials = grpc.credentials.createSsl(lndCert);
-      this.lightning = new LightningClient(`${host}:${port}`, credentials);
+      this.credentials = grpc.credentials.createSsl(lndCert);
 
       const adminMacaroon = fs.readFileSync(macaroonpath);
       this.meta = new grpc.Metadata();
       this.meta.add('macaroon', adminMacaroon.toString('hex'));
-
-      this.setStatus(ClientStatus.CONNECTION_VERIFIED); // TODO: verify connection
+      this.connect();
```

Comment:
I realized upon a second pass that we have an `async` call here within a constructor which is necessarily synchronous. We want to make sure we are connected before we continue with our startup process. Can you make the `connect` method public and move it from the constructor to `Xud.ts` right after we create the client by calling `await this.lndClient.connect();`?